pkg/server/server.go

```diff
@@ -96,19 +96,30 @@ func renderer(templatePath string) multitemplate.Renderer {
 	return r
 }
 
-func requireAuth(db *sqlx.DB, secret string) func(*gin.Context) {
+func jobAuth(ctx *gin.Context, db *sqlx.DB, secret string) (string, error) {
+	jobID := ctx.Param("id")
+	job, err := data.GetJob(jobID, db)
+	if err != nil {
+		log.Println(fmt.Errorf("requireAuth failed to getJob: %w", err))
+		ctx.AbortWithStatus(http.StatusInternalServerError)
+		return "", err
+	}
+
+	return SignatureForJob(job, secret), nil
+}
+
+func requireAuth(db *sqlx.DB, secret string, routeAuthCallback func(*gin.Context, *sqlx.DB, string) (string, error)) func(*gin.Context) {
```

Comment:
In the same vein as the previous comment, maybe this becomes a little more specific as well -- `requireTokenAuth`.

Reply:
Yeah, that's a good call.